pkg/agent/agent.go

```diff
@@ -344,14 +357,20 @@ func startDiscoveryCache(instanceCacheLock *sync.Mutex) {
 }
 
 func startStreamMode(agent agentData) error {
-	cs, err := stream.NewStreamer(
-		api.NewClient(agent.cfg.GetTLSConfig(), agent.cfg.GetProxyURL()),
-		agent.cfg,
-		agent.tokenRequester,
+	proxyHandler := agent.proxyResourceHandler.(handler.Handler)
```

Comment:
Not that big a deal, but instead of doing a type conversion we could have `NewProxyHandler` return the struct which implements both interfaces, and `agent.proxyResourceHandler` would be set to the struct instead of the `ProxyHandler` Interface.

Reply:
Thanks Trevor